addons/survey/models/survey_question.py

```diff
@@ -403,11 +403,11 @@ def validate_question(self, answer, comment=None):
         if isinstance(answer, str):
             answer = answer.strip()
         # Empty answer to mandatory question
-        if self.constr_mandatory and not answer and self.question_type not in ['simple_choice', 'multiple_choice']:
-            return {self.id: self.constr_error_msg or _('This question requires an answer.')}
-
+        if not answer and self.question_type not in ['simple_choice', 'multiple_choice']:
+            if self.constr_mandatory and not self.survey_id.users_can_go_back:
+                return {self.id: self.constr_error_msg or _('This question requires an answer.')}
         # because in choices question types, comment can count as answer
```

Comment:
This comment should be part of the one on top as it the first mention of choices questions is there now.